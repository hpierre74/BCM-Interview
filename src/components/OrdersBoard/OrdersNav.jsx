import React from "react";
import { Nav, NavItem, NavLink } from 'reactstrap';

const energies = {
    "1": "Wind power",
    "2": "Biomass",
    "3": "Solar",
    "4": "Gas"
}

const OrdersNav = props => {
  const renderEnergyNavItem = () => {
    return Object.values(energies).map((energy, index) => {
      return (
        <NavItem>
          <NavLink href="#"> { energy } </NavLink>
        </NavItem>
      );
    });
  };

  return (
    <Nav vertical>
      { renderEnergyNavItem() }
    </Nav>
  );
};

export default OrdersNav;
