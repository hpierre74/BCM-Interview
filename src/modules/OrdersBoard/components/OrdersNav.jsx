import React from "react";
import { Nav, NavItem, NavLink } from 'reactstrap';
import PropTypes from 'prop-types';

const energies = {
    "1": "Wind power",
    "2": "Biomass",
    "3": "Solar",
    "4": "Gas"
}

const OrdersNav = props => {
  const renderEnergyNavItem = () => {
    return Object.values(energies).map((energy, index) => {
      return (
        <NavItem key={index}>
          <NavLink href="#"> { energy } </NavLink>
        </NavItem>
      );
    });
  };

  return (
    <Nav vertical>
      { renderEnergyNavItem() }
    </Nav>
  );
};

OrdersNav.propTypes = {
    energies: PropTypes.object
}

export default OrdersNav;
