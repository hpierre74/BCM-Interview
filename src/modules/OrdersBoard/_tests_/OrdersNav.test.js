import OrdersNav from '../components/OrdersNav';
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';


describe('OrdersNav component renders the children correctly', () => {
    it('renders correctly', () => {  
        const rendered = renderer.create(
        <OrdersNav />
      );
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });
