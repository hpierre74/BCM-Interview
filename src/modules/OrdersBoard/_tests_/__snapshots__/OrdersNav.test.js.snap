// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersNav component renders the children correctly renders correctly 1`] = `
<ul
  className="nav flex-column"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
       
      Wind power
       
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
       
      Biomass
       
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
       
      Solar
       
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
       
      Gas
       
    </a>
  </li>
</ul>
`;
